src/ir/module.h

```diff
@@ -57,9 +57,9 @@ class Operation {
 
   std::string ToString(SsaNames& ssa_names) const;
 
-  template <typename Derived>
-  void Accept(IRVisitor<Derived>& visitor) const {
-    visitor.Visit(*this);
+  template <typename Derived, typename Result=void>
```

Comment:
I don't think you need default `Result` to void here and below. This is always called from a visitor, which will have a `Result`.